### PR TITLE
Support PHP 8

### PR DESCRIPTION
**Pull request by lstrojny**
Support for PHP 8 based on @phanan’s PR https://github.com/lstrojny/functional-php/pull/236

### Todos:

- [x] Upgrade PHPUnit
- [x] Update tests to work with PHPUnit's method changes, removals, and deprecations
- [x] Upgrade PHP-CS-Fixer
	- [x] Temporarily allow PHP-CS-Fixer to fail for PHP~8, as the tool doesn't support it yet
- [x] Change the `AbstractTestCase::expectArgumentError` method to adapt to the new message format thrown in PHP 8 (actually, create a new method altogether).
- [x] Fix the few broken tests, which might come from broken functionalities themselves. @lstrojny maybe can lend a hand here?
- [x] Opt out for named parameters
- [x] Figure out backwards compatibility

### CODE REVIEW

File: src/Functional/PartialAny.php [hunk 1]
```diff
@@ -37,7 +38,8 @@ function partial_any(callable $callback, ...$arguments)
     };
 }
 
-/** @return resource */
+/** @return resource * @no-named-arguments
+ */
```
Comment (phanan) [inline src/Functional/PartialAny.php:42]:
```suggestion
/** 
 * @return resource 
 * @no-named-arguments
 */
```

File: src/Functional/PartialAny.php [hunk 2]
```diff
@@ -50,7 +52,8 @@ function …()
 }
 
 
-/** @return resource */
+/** @return resource * @no-named-arguments
+ */
```
Comment (phanan) [inline src/Functional/PartialAny.php:56]:
```suggestion
/** 
 * @return resource 
 * @no-named-arguments
 */
```

File: tests/Functional/AnnotationsTest.php [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+ * @license   https://opensource.org/licenses/MIT MIT
+ * @link      https://github.com/lstrojny/functional-php
+ */
+
+namespace Functional\Tests;
+
+use ReflectionFunction;
+
+use function Functional\group;
+use function get_defined_functions;
```
Comment (phanan) [inline tests/Functional/AnnotationsTest.php:16]: Do we or do we not import the root namespaced functions? It looks like a mix at the moment.

**Review by phanan — APPROVED** [review]

File: tests/Functional/AnnotationsTest.php (hunk 3 above)
Comment (lstrojny) [reply tests/Functional/AnnotationsTest.php]: It’s a mess, yeah. Functional shows it's age here (I just realized when updating the copyright notice it’s 10 years anniversary). When it started, `use function …` was not a thing. But yes, it should be imported everywhere.

**Comment by lstrojny** [comment]
> Most of these attributes can be removed, as their values are the same as the default.

Good point, done

**Pull request merged**